src/Monad/KleisliCategory.idr

```diff
@@ -0,0 +1,23 @@
+module Monad.KleisliCategory
+
+import Basic.Category
+import Basic.Functor
+import Basic.NaturalTransformation
+import Monad.Monad
+
+%access public export
+%default total
+
+kleisliCategory : (cat : Category) -> UnlawfulMonad cat -> Category
+kleisliCategory cat m =
+  MkCategory
+    (obj cat)
+    (\a, b => mor cat a (mapObj (functor m) b))
+    (\a => component (unit m) a)
+    (\a, b, c, f, g =>
+      compose cat _ _ _
+        (compose cat _ _ _ f (mapMor (functor m) _ _ g))
+        (component (multiplication m) c))
+    ?leftId
```

Comment:
So this is the part that's better done in Agda?

Reply:
Yes, basically. When I push the lawful monads, there'll be a lot more of this. The reason is that Agda's inferrence for implicit arguments is a lot better - in those examples I have right now Agda can infer all of those argumets just fine while Idris fails at maybe 20% or so of them.